### PR TITLE
Hitting someone in melee with the explosive lance now moves you to the same tile as them before it sets off the grenade.

### PR DESCRIPTION
**Pull request by timothymtorres**
From @Iamgoofball PR in https://github.com/tgstation/tgstation/pull/65112

> Closes #65090
> 
> ## About The Pull Request
> Alternative to #65090 Hitting someone in melee with the explosive lance now moves you to the same tile as them before it sets off the spear.
> 
> ## Why It's Good For The Game
> It's possible to make an explosive mix that completely vaporizes the target while leaving all the tiles next to it relatively unharmed or at least in a revivable state. This is working as intended for explosion code. However, when applied to the concept for the explosive lance, it doesn't really mesh with the concept as Kor envisioned it when adding them:
> 
> chrome_Vq5APFuVq9
> 
> As you can see, Kor very much intended for this to be a "if I'm dying you're coming with me" style weapon. It used to be this in the past, and the natural content churn of the game means it isn't anymore. This change should restore the weapon to it's former intended (non-meta) functionality rather than being a powergame weapon.

## Changelog
🆑 Iamgoofball
balance: Hitting someone in melee with the explosive lance now moves you to the same tile as them before it sets off the grenade. 
/🆑

### CODE REVIEW

File: code/game/objects/items/spear.dm [hunk 1]
```diff
@@ -139,7 +139,9 @@
 	if(iseffect(AM)) //and no accidentally wasting your moment of glory on graffiti
 		return
 	user.say("[war_cry]", forced="spear warcry")
-	explosive.forceMove(AM)
+	if(ismob(AM))
+		user.forceMove(get_turf(AM))
```
Comment (Mothblocks) [inline code/game/objects/items/spear.dm:143]: Make this do a normal move so it still factors in any immobilizers you might have, walls, etc. It's fine if it fails.

Comment (Iamgoofball) [reply code/game/objects/items/spear.dm:143]: that won't work cuz it'll proc bump code and make the two people swap places

File: code/game/objects/items/spear.dm [hunk 2]
```diff
@@ -139,7 +139,9 @@
...
+	explosive.forceMove(get_turf(AM))
```
Comment (ATH1909) [inline code/game/objects/items/spear.dm:144]:
```suggestion
	explosive.forceMove(AM)
```
the grenade is supposed to be jabbed into your target, not the tile they're standing on

Comment (Iamgoofball) [reply code/game/objects/items/spear.dm:144]: this change is intentional athath

Comment (ATH1909) [reply code/game/objects/items/spear.dm:144]: why isn't it in the pr body, then?

File: code/game/objects/items/spear.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/items/spear.dm:143]: Make 'em rest or something beforehand then, but either way we shouldn't be forceMove'ing this.

Comment (Mothblocks) [reply code/game/objects/items/spear.dm:143]: image

I'm fine with this answer

Comment (Mothblocks) [reply code/game/objects/items/spear.dm:143]: Also even with Goof's answer still make them rest or something so they don't swap in place

**Comment by derekhearst** [comment]
Good change, elances are too meta and robust at the moment against nukies. It's a suicide weapon not a free gib one.

**Comment by bluetshirtguy** [comment]
nice pr, people shouldnt be trying to game a suicide weapon/

**Comment by timothymtorres** [comment]
RIP 🪦

**Pull request closed without merging**